### PR TITLE
build(release): adds release scripts and empty changelogs

### LINKED ISSUE #1814: Write simple release scripts/processes for the packages
**Why**

To enable the packages to be published publicly.

**How**

This should mostly require piggybacking on lerna and configuring it to automate releases.

**Resolution**

Each package's release process is documented and working.

**Difficulty Score [1-10]**

6: ~1 day

### PR DESCRIPTION
**Pull request by mrice32**
**Motivation**

To add a release process for the packages in the monorepo.

**Summary**

This adds two commands:

1. `yarn run create-release`, which is meant to be run locally to generate the package bumps and the changelogs.
2. `yarn run publish-release`, which is run by ci on each master commit. It will publish to the npm registry and tag releases when a package version changes.

It also adds empty changelogs.

I ran a bunch of test publishes on pre-release versions. So, I've set the version of all packages to 1.0.0-alpha.11 (> all the test releases). Merging this PR will run a release, but I'd like to vet the entire flow before publishing a real (non-alpha) 1.0.0 version with real changelogs, etc.

**Details**

Feel free to skip the following if you're not interested in _why_ the implementation came out this way :).

Because lerna is not really set up to construct releases locally and have them published in ci, there is a little bit of awkwardness in this implementation. Generally, lerna wants to do all of the change detection, release construction, version bumping, git tagging, etc in one cohesive step. However, since we'd like to do the tagging (and publishing) in ci _after_ the version numbers have been bumped and the changelogs have been generated, we have to do the tagging manually. This also means we can't get it to automatically publish github releases.

Similarly, because lerna generally uses git tags as its source of truth by default, we pass an option to force it to use the npm registry as its source of truth so it will 1) _only_ publish in that step (rather than reconstructing the entire release) and 2) be self-consistent and only use its previous releases as its source of truth rather than our exogenously constructed git tags.

**Issue(s)**

Fixes #1814.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,12 +4,15 @@
...
   "scripts": {
+    "bootstrap": "npx lerna bootstrap",
     "lint": "npm run eslint && npm run prettier -- --list-different",
     "lint-fix": "npm run eslint -- --fix && npm run prettier -- --write",
     "eslint": "eslint './**/*.js'",
     "prettier": "prettier './**/*.js' './**/*.sol' './**/*.md'",
     "test": "lerna run --stream --concurrency=1 test",
-    "load-addresses": "yarn workspace @umaprotocol/core load-addresses"
+    "load-addresses": "yarn workspace @umaprotocol/core load-addresses",
+    "publish": "lerna publish",
```
Comment (nicholaspai) [inline package.json:14]: Does this publish all packages at once?

**Comment by daywiss** [comment]
is this auto generated?

**Comment by mrice32** [comment]
> is this auto generated?

Yeah, this is basically my testing ground for getting the releases working right.

**Comment by daywiss** [comment]
just want to comment that it worries me from your description that it sounds like we are fighting the way lerna wants to work. i dont really have experience here or a suggestion, but if you are confident that this approach is allowed without too much trouble in lerna then I think that is fine. 

what does lernas recommended release process look like and why does that not work in our situation?

**Comment by mrice32** [comment]
> just want to comment that it worries me from your description that it sounds like we are fighting the way lerna wants to work. i dont really have experience here or a suggestion, but if you are confident that this approach is allowed without too much trouble in lerna then I think that is fine.
> 
> what does lernas recommended release process look like and why does that not work in our situation?

This is a good point. Their recommended approach is to do direct pushes to master on release with no PRs or review process. We could do that, but it would mean that we have some commits that don't go through PRs.

I've done a decent amount of reading on lerna, and I think they recognize that they are working against the way the industry is moving (to have mechanical releases done entirely by ci).

Some relevant issues:

https://github.com/lerna/lerna/issues/653
https://github.com/lerna/lerna/issues/2162
https://github.com/lerna/lerna/issues/1385 -- specifically this comment: https://github.com/lerna/lerna/issues/1385#issuecomment-383159494.

Totally understand your worry. I think what we have in this PR is _good enough_, but may need to be changed down the road as we fall into pitfalls. I'm not married to the solution, but right now, I don't see another good way to make this work other than 1) changing our dev process so that we allow direct pushes to master without ci or review OR using a different tool.

**Comment by mrice32** [comment]
> So what commands will we need to run before merging in PR's to get releases run properly? just trying to synthesize these changes

You'll run `yarn run create-release` locally from root, which will walk you through bumping the versions and automatically generate the changelogs based on the conventional commits since the last git tags in the repo. You might want to clean up the changelogs or manually modify versions if you don't like the autogenerated ones. Whenever you're ready, you can publish a PR with this information and ci will do the release automatically.

**Comment by daywiss** [comment]
thanks for the context. I feel better knowing that this is something that they know is an issue. im fine with trying it out as you have designed it!

**Pull request merged**